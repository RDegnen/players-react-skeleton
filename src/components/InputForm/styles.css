.formCol {
  padding-left: 30px; 
}