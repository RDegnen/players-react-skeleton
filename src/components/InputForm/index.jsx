import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Form, FormGroup, Label, Input, Col, Row } from 'reactstrap';

import componentProps from '../props';
import './styles.css';

const InputForm = ({ formGroups, handleInput, handleSubmit }) => (
  <Row>
    <Col m="6" className="formCol">
      <Form onSubmit={handleSubmit}>
        {formGroups.map(({
          field, type, id, label, renderGroup,
        }) => (
          <Fragment key={field}>
            {renderGroup ? (
              renderGroup()
            ) : (
              <FormGroup key={`${field}Group`} row>
                <Label key={`${field}Label`} for={field}>{label}</Label>
                <Input
                  key={`${field}Input`}
                  type={type}
                  name={field}
                  id={id}
                  placeholder={label}
                  onChange={handleInput}
                />
              </FormGroup>
            )}
          </Fragment>
        ))}
      </Form>
    </Col>
    <Col m="6" />
  </Row>
);

InputForm.propTypes = {
  formGroups: componentProps.formGroupShape.isRequired,
  handleInput: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
};

export default InputForm;
