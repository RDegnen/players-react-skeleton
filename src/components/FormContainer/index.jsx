import React from 'react';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';

import InputForm from '../../components/InputForm';
import componentProps from '../props';

class FormContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleInput = this.handleInput.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleInput({ target }) {
    const { name, value } = target;
    const numberInputs = ['rating'];
    this.setState({ [name]: numberInputs.includes(name) ? parseInt(value, 10) : value });
  }

  handleSubmit(event) {
    event.preventDefault();
    const { history, submitAction } = this.props;
    submitAction(this.state, history);
  }

  render() {
    const { formGroups } = this.props;
    return (
      <InputForm
        formGroups={formGroups}
        handleInput={this.handleInput}
        handleSubmit={this.handleSubmit}
      />
    );
  }
}

FormContainer.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
  formGroups: componentProps.formGroupShape.isRequired,
  submitAction: PropTypes.func.isRequired,
};

export default FormContainer;
