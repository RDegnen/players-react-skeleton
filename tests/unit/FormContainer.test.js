import React from 'react';
import { shallow } from 'enzyme';

import FormContainer from '../../src/components/FormContainer';
import InputForm from '../../src/components/InputForm';

const defaultProps = {
  formGroups: [
    {
      field: 'email',
      type: 'email',
      id: 'email',
      label: 'Email',
    },
    {
      field: 'password',
      type: 'password',
      id: 'password',
      label: 'Password',
    },
    {
      renderGroup: jest.fn(),
    },
  ],
  submitAction: jest.fn(),
  history: {},
};

const getComponent = props => shallow(<FormContainer {...defaultProps} {...props} />);

describe('<FormContainer />', () => {
  it('should match snapshot', () => {
    const component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('should render and InputForm', () => {
    const component = getComponent();
    expect(component.find(InputForm).length).toBe(1);
  });
});
