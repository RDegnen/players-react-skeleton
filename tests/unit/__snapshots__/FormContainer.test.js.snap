// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormContainer /> should match snapshot 1`] = `
<InputForm
  formGroups={
    Array [
      Object {
        "field": "email",
        "id": "email",
        "label": "Email",
        "type": "email",
      },
      Object {
        "field": "password",
        "id": "password",
        "label": "Password",
        "type": "password",
      },
      Object {
        "renderGroup": [MockFunction],
      },
    ]
  }
  handleInput={[Function]}
  handleSubmit={[Function]}
/>
`;
